### ISSUE TITLE
dr_wav: untrusted value as argument in drwav_open_and_read_pcm_frames_s16

### ISSUE THREAD

**Issue by krcroft** [mackron/dr_libs#84]
```
4765 drwav_int16* drwav_open_and_read_pcm_frames_s16(drwav_read_proc onRead, drwav_seek_proc onSeek, void* pUserData, unsigned int* channelsOut, unsigned int* sampleRateOut, drwav_uint64* totalFrameCountOut, const drwav_allocation_callbacks* pAllocationCallbacks)
4766{
4767    drwav wav;
4768
    	1. Condition channelsOut, taking true branch.
4769    if (channelsOut) {
4770        *channelsOut = 0;
4771    }
    	2. Condition sampleRateOut, taking true branch.
4772    if (sampleRateOut) {
4773        *sampleRateOut = 0;
4774    }
    	3. Condition totalFrameCountOut, taking true branch.
4775    if (totalFrameCountOut) {
4776        *totalFrameCountOut = 0;
4777    }
4778
    	4. tainted_data_argument: Calling function drwav_init taints argument wav.channels. [show details]
    	5. Condition !drwav_init(&wav, onRead, onSeek, pUserData, pAllocationCallbacks), taking false branch.
4779    if (!drwav_init(&wav, onRead, onSeek, pUserData, pAllocationCallbacks)) {
4780        return NULL;
4781    }
4782
    	
CID 277167 (#1 of 1): Untrusted value as argument (TAINTED_SCALAR)
6. tainted_data: Passing tainted variable wav.channels to a tainted sink. [show details]
4783    return drwav__read_pcm_frames_and_close_s16(&wav, channelsOut, sampleRateOut, totalFrameCountOut);
4784}
```

**Comment by mackron** [comment]
I'm not 100% sure what this report means by "tainted", but what I'm suspecting it may be related to is how I don't sanitize it with an upper bound check, and then use it as part of a memory allocation later. Not totally sure about this, but I've pushed a possible fix here: 69dde2b3a02aa3e355a4125dd3b2d344acb88398.

**Comment by krcroft** [comment]
That was it; confirmed fixed in the latest scan. Thank you @mackron !
